### ISSUE TITLE
Build trouble with setup.py install

### ISSUE THREAD

**Issue by stuaxo** [jparyani/pycapnp#54]
I haven't managed to install this with pip, so have been trying with a clone of the git repo - here's what happens:
```
$ python setup.py install
missing cimport in module 'capnp_cpp': ./capnp/includes/schema_cpp.pxd
missing cimport in module 'schema_cpp': ./capnp/includes/capnp_cpp.pxd
missing cimport in module 'non_circular': ./capnp/helpers/helpers.pxd
Compiling capnp/lib/capnp.pyx because it changed.
Cythonizing capnp/lib/capnp.pyx

Error compiling Cython file:
------------------------------------------------------------
...
    cdef _init(self, capnp.DynamicEnum other, object parent):
        self.thisptr = other
        self._parent = parent
        return self

    cpdef _as_str(self) except +reraise_kj_exception:
         ^
------------------------------------------------------------

capnp/lib/capnp.pyx:877:10: Signature not compatible with previous declaration

Error compiling Cython file:
------------------------------------------------------------
...
cdef class _DynamicEnum:
    cdef capnp.DynamicEnum thisptr
    cdef public object _parent

    cdef _init(self, capnp.DynamicEnum other, object parent)
    cpdef _as_str(self)
                 ^
------------------------------------------------------------

capnp/lib/capnp.pxd:105:18: Previous declaration is here

Error compiling Cython file:
------------------------------------------------------------
...
        """
        self._check_write()
        _write_packed_message_to_fd(file.fileno(), self._parent)
        self._is_written = True

    cpdef to_bytes(_DynamicStructBuilder self) except +reraise_kj_exception:
         ^
------------------------------------------------------------

capnp/lib/capnp.pyx:1143:10: Signature not compatible with previous declaration

Error compiling Cython file:
------------------------------------------------------------
...
    cdef object _schema

    cdef _init(self, DynamicStruct_Builder other, object parent, bint isRoot=?, bint tryRegistry=?)

    cdef _check_write(self)
    cpdef to_bytes(_DynamicStructBuilder self)
                  ^
------------------------------------------------------------

capnp/lib/capnp.pxd:55:19: Previous declaration is here

Error compiling Cython file:
------------------------------------------------------------
...
        cdef const char* ptr = <const char *>array.begin()
        cdef bytes ret = ptr[:8*array.size()]
        self._is_written = True
        return ret

    cpdef _to_bytes_packed_helper(_DynamicStructBuilder self, word_count) except +reraise_kj_exception:
         ^
------------------------------------------------------------

capnp/lib/capnp.pyx:1160:10: Signature not compatible with previous declaration

Error compiling Cython file:
------------------------------------------------------------
...

    cdef _init(self, DynamicStruct_Builder other, object parent, bint isRoot=?, bint tryRegistry=?)

    cdef _check_write(self)
    cpdef to_bytes(_DynamicStructBuilder self)
    cpdef _to_bytes_packed_helper(_DynamicStructBuilder self, word_count)
                                 ^
------------------------------------------------------------

capnp/lib/capnp.pxd:56:34: Previous declaration is here

Error compiling Cython file:
------------------------------------------------------------
...
        array = helpers.messageToPackedBytes(deref(builder.thisptr), word_count)
        cdef const char* ptr = <const char *>array.begin()
        cdef bytes ret = ptr[:array.size()]
        return ret

    cpdef to_bytes_packed(_DynamicStructBuilder self) except +reraise_kj_exception:
         ^
------------------------------------------------------------

capnp/lib/capnp.pyx:1167:10: Signature not compatible with previous declaration

Error compiling Cython file:
------------------------------------------------------------
...
    cdef _init(self, DynamicStruct_Builder other, object parent, bint isRoot=?, bint tryRegistry=?)

    cdef _check_write(self)
    cpdef to_bytes(_DynamicStructBuilder self)
    cpdef _to_bytes_packed_helper(_DynamicStructBuilder self, word_count)
    cpdef to_bytes_packed(_DynamicStructBuilder self)
                         ^
------------------------------------------------------------

capnp/lib/capnp.pxd:57:26: Previous declaration is here
Traceback (most recent call last):
  File "setup.py", line 130, in <module>
    extensions = cythonize('capnp/lib/*.pyx')
  File "/mnt/data/home/stu/.virtualenvs/tmp/local/lib/python2.7/site-packages/Cython/Build/Dependencies.py", line 865, in cythonize
    cythonize_one(*args[1:])
  File "/mnt/data/home/stu/.virtualenvs/tmp/local/lib/python2.7/site-packages/Cython/Build/Dependencies.py", line 985, in cythonize_one
```

**Comment by jparyani** [comment]
Unfortunately this is due to an upstream bug in cython v0.22. I've commented on their mailing list about it, but have gotten no response (https://groups.google.com/forum/#!topic/cython-users/Yt-qBJroZmg).

I'm not disabling cython by default for now (fixed in b2daf3911dd0e749df3557eff426671840ade7d1). If you're using the git repo directly and need to run cython, you can do so with the --force-cython flag (ie. `python setup.py install --force-cython` or `pip install --install-option '--force-cython'`

**Comment by stuaxo** [comment]
OK, so now I get the error below - and a different one installing from pip - should I make a different bug for the pip install one ?
```
tion: 'entry_points'
  warnings.warn(msg)
/usr/lib/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'install_requires'
  warnings.warn(msg)
running install
running build
running build_py
creating build
creating build/lib.linux-x86_64-2.7
creating build/lib.linux-x86_64-2.7/capnp
copying capnp/_gen.py -> build/lib.linux-x86_64-2.7/capnp
copying capnp/version.py -> build/lib.linux-x86_64-2.7/capnp
copying capnp/__init__.py -> build/lib.linux-x86_64-2.7/capnp
copying capnp/__init__.pxd -> build/lib.linux-x86_64-2.7/capnp
copying capnp/schema.capnp -> build/lib.linux-x86_64-2.7/capnp
copying capnp/c++.capnp -> build/lib.linux-x86_64-2.7/capnp
creating build/lib.linux-x86_64-2.7/capnp/helpers
copying capnp/helpers/helpers.pxd -> build/lib.linux-x86_64-2.7/capnp/helpers
copying capnp/helpers/__init__.pxd -> build/lib.linux-x86_64-2.7/capnp/helpers
copying capnp/helpers/non_circular.pxd -> build/lib.linux-x86_64-2.7/capnp/helpers
copying capnp/helpers/checkCompiler.h -> build/lib.linux-x86_64-2.7/capnp/helpers
copying capnp/helpers/capabilityHelper.h -> build/lib.linux-x86_64-2.7/capnp/helpers
copying capnp/helpers/asyncHelper.h -> build/lib.linux-x86_64-2.7/capnp/helpers
copying capnp/helpers/rpcHelper.h -> build/lib.linux-x86_64-2.7/capnp/helpers
copying capnp/helpers/fixMaybe.h -> build/lib.linux-x86_64-2.7/capnp/helpers
copying capnp/helpers/serialize.h -> build/lib.linux-x86_64-2.7/capnp/helpers
creating build/lib.linux-x86_64-2.7/capnp/includes
copying capnp/includes/types.pxd -> build/lib.linux-x86_64-2.7/capnp/includes
copying capnp/includes/capnp_cpp.pxd -> build/lib.linux-x86_64-2.7/capnp/includes
copying capnp/includes/__init__.pxd -> build/lib.linux-x86_64-2.7/capnp/includes
copying capnp/includes/schema_cpp.pxd -> build/lib.linux-x86_64-2.7/capnp/includes
creating build/lib.linux-x86_64-2.7/capnp/lib
copying capnp/lib/__init__.pxd -> build/lib.linux-x86_64-2.7/capnp/lib
copying capnp/lib/capnp.pxd -> build/lib.linux-x86_64-2.7/capnp/lib
copying capnp/lib/__init__.py -> build/lib.linux-x86_64-2.7/capnp/lib
copying capnp/lib/pickle_helper.py -> build/lib.linux-x86_64-2.7/capnp/lib
copying capnp/lib/capnp.pyx -> build/lib.linux-x86_64-2.7/capnp/lib
creating build/lib.linux-x86_64-2.7/capnp/templates
copying capnp/templates/setup.py.tmpl -> build/lib.linux-x86_64-2.7/capnp/templates
copying capnp/templates/module.pyx -> build/lib.linux-x86_64-2.7/capnp/templates
running build_ext
creating /tmp/tmpU6k0c4/tmp
cc -c /tmp/timer_create8rNHq1.c -o /tmp/tmpU6k0c4/tmp/timer_create8rNHq1.o
cc /tmp/tmpU6k0c4/tmp/timer_create8rNHq1.o -o /tmp/tmpU6k0c4/a.out
/tmp/tmpU6k0c4/tmp/timer_create8rNHq1.o: In function `main':
timer_create8rNHq1.c:(.text+0x15): undefined reference to `timer_create'
collect2: error: ld returned 1 exit status
creating tmp/tmpU6k0c4
cc -c /tmp/tmpU6k0c4/vers.cpp -o tmp/tmpU6k0c4/vers.o --std=c++11
cc tmp/tmpU6k0c4/vers.o -lrt -o /tmp/tmpU6k0c4/vers
building 'capnp.lib.capnp' extension
creating build/temp.linux-x86_64-2.7
creating build/temp.linux-x86_64-2.7/capnp
creating build/temp.linux-x86_64-2.7/capnp/lib
x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fno-strict-aliasing -D_FORTIFY_SOURCE=2 -g -fstack-protector-strong -Wformat -Werror=format-security -fPIC -I. -I/usr/include/python2.7 -c capnp/lib/capnp.cpp -o build/temp.linux-x86_64-2.7/capnp/lib/capnp.o --std=c++11
cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
capnp/lib/capnp.cpp:1:2: error: #error Do not use this file, it is the result of a failed Cython compilation.
 #error Do not use this file, it is the result of a failed Cython compilation.
  ^
error: command 'x86_64-linux-gnu-gcc' failed with exit status 1

```

**Comment by jparyani** [comment]
You first error is from a problem with the latest version of Cython (v0.22). There will be a fix whenever v0.22.1 is released, but for now you can get it building with v0.21.2 or lower.

The second error is because the first cythonization failed, and you need to manually clean out the artifacts. `python setup.py clean` should work.

**Comment by stuaxo** [comment]
Cheers - I'll wait for Cython 0.22.1 and give this another go.
